utils/appliance/endpoints/ui.py

```diff
@@ -23,6 +68,52 @@ def __init__(self, owner):
         self.owner = owner
         self.menu = Menu()
 
+    @cached_property
+    def widgetastic(self):
+        """This gives us a widgetastic browser."""
+        return MiqBrowser(manager.ensure_open(), self)
+
+    def create_view(
+            self, view_class, navigate=False, require_navigate=False, additional_context=None):
```

Comment:
So the idea about the integration into the new navigation framework, is that the view will have a "destination" associated with it, and the ```additional_context``` would be replaced by a ```object``` reference. This reference would then be used down on 113 so it would become, ```navigate.navigate(object, destination)```